Library/Homebrew/cmd/fetch.rb

```diff
@@ -167,6 +167,11 @@ def self.fetch
           SimulateSystem.with os: os, arch: arch do
             cask = Cask::CaskLoader.load(ref)
 
+            if cask.depends_on.macos&.satisfied? == false
+              opoo "#{cask.token}: #{cask.depends_on.macos.message(type: :cask)}"
+              next
+            end
```

Comment:
Command: `brew fetch gpgfrontend`

Before:
```
Error: undefined method `specs' for nil:NilClass
/usr/local/Homebrew/Library/Homebrew/cask/download.rb:27:in `url'
/usr/local/Homebrew/Library/Homebrew/cask/download.rb:104:in `determine_url'
/usr/local/Homebrew/Library/Homebrew/downloadable.rb:126:in `determine_url_mirrors'
/usr/local/Homebrew/Library/Homebrew/downloadable.rb:74:in `downloader'
/usr/local/Homebrew/Library/Homebrew/downloadable.rb:50:in `cached_download'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:232:in `fetch_fetchable'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:197:in `fetch_cask'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:174:in `block (3 levels) in fetch'
/usr/local/Homebrew/Library/Homebrew/simulate_system.rb:29:in `with'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:167:in `block (2 levels) in fetch'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:164:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:164:in `block in fetch'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:94:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:94:in `fetch'
/usr/local/Homebrew/Library/Homebrew/brew.rb:94:in `<main>'
```
(without the Sorbet stack trace lines)

After:
```
Warning: gpgfrontend: This software does not run on macOS versions older than Big Sur.
```